dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/validation/Validator.java

```diff
@@ -42,11 +48,30 @@
      */
     void validate( ValidationErrorReporter reporter, TrackerBundle bundle, T input );
 
+    /**
+     * Indicates whether this {@link Validator} should be run given the
+     * {@link TrackerImportStrategy}.
+     *
+     * @param strategy import strategy
+     * @return true if this validator should be run and false otherwise
+     */
     default boolean needsToRun( TrackerImportStrategy strategy )
```

Comment:
Same as my previous comment in another PR, I do think this should be named "shouldRunForStrategy" or similar; Needs to run sounds more like it should always run, or else it should only run sometimes, and it's not related to strategy (Unless you read the javadoc now)

Reply:
its on my todo list, I'll suggest something next week 😄